src/libraries/System.Runtime/tests/System/ByteTests.GenericMath.cs

```diff
@@ -0,0 +1,1175 @@
+            Assert.Equal((byte)0xFF, NumberHelper<byte>.CreateTruncating<ulong>(0x7FFFFFFFFFFFFFFF));
+            Assert.Equal((byte)0x00, NumberHelper<byte>.CreateTruncating<ulong>(0x8000000000000000));
+            Assert.Equal((byte)0xFF, NumberHelper<byte>.CreateTruncating<ulong>(0xFFFFFFFFFFFFFFFF));
+        }
+
+        [Fact]
+        public static void CreateTruncatingFromUIntPtrTest()
+        {
+            if (Environment.Is64BitProcess)
+            {
+                Assert.Equal((byte)0x00, NumberHelper<byte>.CreateTruncating<nuint>(unchecked((nuint)0x0000000000000000)));
+                Assert.Equal((byte)0x01, NumberHelper<byte>.CreateTruncating<nuint>(unchecked((nuint)0x0000000000000001)));
+                Assert.Equal((byte)0xFF, NumberHelper<byte>.CreateTruncating<nuint>(unchecked((nuint)0x7FFFFFFFFFFFFFFF)));
+                Assert.Equal((byte)0x00, NumberHelper<byte>.CreateTruncating<nuint>(unchecked((nuint)0x8000000000000000)));
+                Assert.Equal((byte)0xFF, NumberHelper<byte>.CreateTruncating<nuint>(unchecked((nuint)0xFFFFFFFFFFFFFFFF)));
+            }
+            else
+            {
+                Assert.Equal((byte)0x00, NumberHelper<byte>.CreateTruncating<nuint>((nuint)0x00000000));
+                Assert.Equal((byte)0x01, NumberHelper<byte>.CreateTruncating<nuint>((nuint)0x00000001));
+                Assert.Equal((byte)0xFF, NumberHelper<byte>.CreateTruncating<nuint>((nuint)0x7FFFFFFF));
+                Assert.Equal((byte)0x00, NumberHelper<byte>.CreateTruncating<nuint>((nuint)0x80000000));
+                Assert.Equal((byte)0xFF, NumberHelper<byte>.CreateTruncating<nuint>((nuint)0xFFFFFFFF));
+            }
+        }
+
+        [Fact]
+        public static void DivRemTest()
+        {
+            Assert.Equal(((byte)0x00, (byte)0x00), NumberHelper<byte>.DivRem((byte)0x00, (byte)2));
+            Assert.Equal(((byte)0x00, (byte)0x01), NumberHelper<byte>.DivRem((byte)0x01, (byte)2));
+            Assert.Equal(((byte)0x3F, (byte)0x01), NumberHelper<byte>.DivRem((byte)0x7F, (byte)2));
+            Assert.Equal(((byte)0x40, (byte)0x00), NumberHelper<byte>.DivRem((byte)0x80, (byte)2));
+            Assert.Equal(((byte)0x7F, (byte)0x01), NumberHelper<byte>.DivRem((byte)0xFF, (byte)2));
+        }
+
+        [Fact]
+        public static void MaxTest()
+        {
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Max((byte)0x00, (byte)1));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Max((byte)0x01, (byte)1));
+            Assert.Equal((byte)0x7F, NumberHelper<byte>.Max((byte)0x7F, (byte)1));
+            Assert.Equal((byte)0x80, NumberHelper<byte>.Max((byte)0x80, (byte)1));
+            Assert.Equal((byte)0xFF, NumberHelper<byte>.Max((byte)0xFF, (byte)1));
+        }
+
+        [Fact]
+        public static void MinTest()
+        {
+            Assert.Equal((byte)0x00, NumberHelper<byte>.Min((byte)0x00, (byte)1));
```

Comment:
I feel like Theory would be easier here, but it's a nit, so I'm not too bothered. I also don't know how you generated these. If you used a tt file, I'm sure writing a `Fact` was easier

Reply:
Oh never mind, I saw your reply to Toub